ui/store/actions.js

```diff
@@ -1317,6 +1317,43 @@ export function addCollectible(address, tokenID, dontShowLoadingIndicator) {
   };
 }
 
+export function addCollectibleVerifyOwnership(
+  address,
+  tokenID,
+  dontShowLoadingIndicator,
+) {
+  return async (dispatch) => {
+    if (!address) {
+      throw new Error('MetaMask - Cannot add collectible without address');
+    }
+    if (!tokenID) {
+      throw new Error('MetaMask - Cannot add collectible without tokenID');
+    }
+    if (!dontShowLoadingIndicator) {
+      dispatch(showLoadingIndication());
+    }
+    try {
+      await promisifiedBackground.addCollectibleVerifyOwnership(
+        address,
+        tokenID,
+      );
+    } catch (error) {
+      if (
+        error.message.includes('This collectible is not owned by the user') ||
```

Comment:
These look like errors that we would want to display instead of throw? And any other errors would be the ones we want to throw. If not, can you explain why we are throwing these errors and displaying all others?

Reply:
So perhaps this isn't the best pattern but let me explain how its working:

These two errors are the expected ones, that I forward along to the UI [here](https://github.com/MetaMask/metamask-extension/blob/f54e3747aaaccba1e84ecf1ae6b5994e17079131/ui/pages/add-collectible/add-collectible.component.js#L28) and ultimately show them in a toast indicating that the collectible was not added.

Other unexpected errors are logged and displayed in the way we display (unknown/unexpected?) errors for every other action.

Reply:
okay, I get it, I think this is fine